stub_uploader/metadata.py

```diff
@@ -350,9 +350,11 @@ def verify_external_req_stubs_require_its_runtime(
 
     runtime_req_name = EXTERNAL_RUNTIME_REQ_MAP.get(req.name, req.name)
 
-    if not runtime_in_upstream_requires(
-        req, data
-    ) and not runtime_in_upstream_sdist_requires(req, data):
+    if (
+        runtime_req_name != upstream_distribution  # Allow `types-foo` to require `foo`
```

Comment:
I think this is the bug that I alluded to when I wrote the refactoring PR. But wouldn't it be clearer to just check `req.name == upstream_distribution` at the beginning of the function and exit early? This would also skip fetching the data from PyPI.

Reply:
If we don't mind not validating that `upstream_distribution` does exist on pypi

Reply:
> I think this is the bug that I alluded to when I wrote the refactoring PR. But wouldn't it be clearer to just check `req.name == upstream_distribution`

I'm not sure which side the the `EXTERNAL_RUNTIME_REQ_MAP` makes sense here. But I was reusing what's in `runtime_in_upstream_sdist_requires` and `runtime_in_upstream_requires`